### PR TITLE
Fix E2E test cleanup errors

### LINKED ISSUE #1485: E2E test cleanup produces errors
The cleanup process of the E2E test runner does not always succeed due to the following reasons:
- Operators get deleted before the managed namespace, leaving the managed namespace stuck in the `TERMINATING` state because the finalizers cannot be executed.
- Namespace operators are currently only deployed to a single namespace but the loop in the manifest template does not take that into consideration -- which results in duplicate manifests being generated for the same namespace. Passing these manifests to `kubectl delete` produces a conflict error:
```
Error from server (Conflict): error when deleting "/tmp/eck-e2e/e2e-2b7ts/config_e2e_operator_namespaces.yaml": Operation cannot be fulfilled on namespaces "e2e-2b7ts-ns-operators": The system is ensuring all content is removed from this namespace.  Upon completion, this namespace will automatically be purged by the system.
```

### PR DESCRIPTION
**Pull request by charith-elastic**
Deduplicates manifests and re-orders the deletion of resources to avoid error messages during the test cleanup phase.

Fixes #1485

### CODE REVIEW

**Comment by charith-elastic** [comment]
retest this please

File: operators/test/e2e/cmd/run/run.go [hunk 1]
```diff
@@ -422,6 +432,15 @@ func (h *helper) renderTemplate(templatePath string, param interface{}) (string,
 	return outFilePath, nil
 }
 
+func (h *helper) deleteResources(file string) func() {
+	return func() {
+		log.Info("Deleting resources", "file", file)
+		if err := h.kubectl("delete", "--all", "--wait", "-f", file); err != nil {
```
Comment (thbkrkr) [inline operators/test/e2e/cmd/run/run.go]: What is the purpose of `--all` when we use `-f file`?

Comment (charith-elastic) [reply operators/test/e2e/cmd/run/run.go]: My understanding is that it should trigger the deletion of uninitialized resources as well. I am not sure whether the `-f` flag overrides that behaviour though.

Comment (thbkrkr) [reply operators/test/e2e/cmd/run/run.go]: From my experience, `-f` seems to delete uninitialized resources but in the doubt I'm fine to keep `--all`.

**Pull request merged**